torch/onnx/utils.py

```diff
@@ -137,23 +132,45 @@ def _export(model, args, f, export_params=True, verbose=False, training=False,
         raise RuntimeError("state_dict changed after running the tracer; "
                            "something weird is happening in your model!")
 
-    trace.set_graph(_optimize_graph(trace.graph(), aten))
+    return trace.graph(), torch_out
+
+
+# NOTE: the output `torch_out` will contain the output tensors resulting from
+# the trace of a Module. In the case that a torch.nn.ScriptModule is passed in,
+# this output will be None, since we are not doing any tracing but rather
+# directly extracting the graph.
+def _export(model, args, f, export_params=True, verbose=False, training=False,
+            input_names=None, output_names=None, aten=False, export_type=ExportTypes.PROTOBUF_FILE):
+    # Special case for common case of passing a single Variable
+    if isinstance(args, torch.autograd.Variable):
+        args = (args, )
+
+    if isinstance(model, torch.jit.ScriptModule):
+        torch_out = None
+        try:
+            graph = model.__getattr__('forward').graph()
+            graph.propagate_shapes(args, False)
```

Comment:
I am pretty sure that `graph` is a graph of a script method, meaning it has extra inputs for all of the parameters it uses. So simply passing `args` here is not correct, it also needs to gather the parameters from the model.

Reply:
@zdevito Ah, yeah I made those fixes on the follow-up PR for the control flow export. Should I squash the two? That seems like the easiest route forward